src/lib/operators/table_scan/abstract_dereferenced_column_table_scan_impl.cpp

```diff
@@ -71,6 +76,20 @@ void AbstractDereferencedColumnTableScanImpl::_scan_reference_segment(const Refe
       matches[match_idx].chunk_offset = sub_pos_list.original_positions[matches[match_idx].chunk_offset];
```

Comment:
Can we debug assert here, that matches is sufficiently allocated (check for capacity)? It looks dangerous. :D

Reply:
I don't see the issue since the loop runs to matches.size().
`sub_pos_list.original_positions[matches[match_idx].chunk_offset]` looks more dangerous to me :D

Reply:
Argh ... I totally misread the lines due to effin Github interface. Never mind!

Can you please extract the `size()` call from the loop?

Reply:
Ofc.
Would you like to DebugAssert that `sub_pos_list.original_positions` is at least `matches[match_idx].chunk_offset` large?